src/config/cli_config.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <boost/program_options.hpp>
+#include <string>
+
+#include "config/config.hpp"
+
+namespace taraxa {
+
+class CliConfig {
+ public:
+  CliConfig(int argc, const char* argv[], const std::string& version);
+
+  ~CliConfig() {}
+
+  // Returns true if node configuration is loaded successfully and command is node
+  bool nodeConfigured();
+
+  // Retrieves loaded node configuration
+  std::shared_ptr<FullNodeConfig> getNodeConfiguration();
+
+  enum NetworkIdType { Mainnet = 1, Testnet, Devnet };
+
+ protected:
+  static constexpr int DEFAULT_NETWORK_ID = Testnet;
+
+  boost::program_options::options_description main_options;
+  boost::program_options::options_description node_command_options;
+  boost::program_options::options_description allowed_options;
+
+  std::string config;
+  std::string wallet;
+  std::string conf_chain;
+  int network_id;
+  std::string data_dir;
+  std::vector<std::string> command;
+  bool boot_node = false;
+  bool destroy_db = 0;
+  bool rebuild_network = 0;
+  bool rebuild_db = 0;
+  bool version = false;
+  uint64_t rebuild_db_period = 0;
+  uint64_t revert_to_period = 0;
+  std::shared_ptr<FullNodeConfig> node_config;
```

Comment:
what is the benefit/need for node_config to be a shared_ptr instead of just FullNodeConfig ?